Polkadot status up to Decoded 2023

filippoweb3 opened the pull request:
This covers just the Polkadot 1.0 part. Another PR will focus on Polkadot 2.0, as well as what is Web3 and how web3 app are structured.

Commit:
added new vision page

Commit:
Radha's suggestion

Commit:
Merge branch 'master' into polkadot-vision

Commit:
Merge branch 'master' into polkadot-vision

Commit:
added Pokadot 1.0

Commit:
preliminary grammar check

Commit:
more info about Polkadot 1.0

Commit:
grammar

Commit:
minor edits

Commit:
added blockspace

Commit:
added decentralization

Commit:
grammar check

Commit:
minor edits

filippoweb3 commented:
@DrW3RK @CrackTheCode016 I would like to request a preliminary review of the Polkadot Vision - Polkadot 1.0. This should be the current state with some thinking around blockspace, decentralization and unstoppability (that I will rename resistance as suggested by Gavin during the talk). Thanks in advance!

Commit:
resilience

Commit:
additions Pokadot 1.0

Commit:
minor edits intro

Commit:
added notes

Commit:
reorg content 1.0

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+Since the release of Bitcoin in 2009, blockchain projects increased exponentially to the order of
+tens of thousands. Different projects have different value propositions, suggesting that the future
+will likely be multi-chain and that inter-chain communication will be crucial to establish
+collaborations and leveraging each other strengths.
```
```suggestion
Since the release of Bitcoin in 2009, blockchain projects increased exponentially to the order of
tens of thousands. Different projects have different value propositions, suggesting that the future
will be multi-chain and that inter-chain communication will be crucial to establish
collaborations and leveraging each other strengths.
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+resilience and blockspace.
+
+Polkadot is a Layer-0 blockchain that brings to the multi-chain vision the following innovative
+concepts and initiatives:
```
```suggestion
Polkadot is a Layer-0 blockchain that brings to the multi-chain vision the following innovations and initiatives:
```

Commit:
finalized 1.0

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+- **Application-specific Layer-1 (L1) blockchains (or parachains).** Polkadot is a sharded network
+  where data are processed in parallel within each shard. Polkadot shards can be heterogenous (i.e.
+  they do not need to be the exact copy of the original shards, like in Ethereum shards). This
+  allows to build L1 chains designed explicitly around their application and value proposition.
```
```suggestion
- **Application-specific Layer-1 (L1) blockchains (or parachains).** Polkadot is a sharded network
  where transactions are processed parallelly within each shard. Polkadot shards can be heterogenous (i.e.
  they do not need the same state transition function as in the proposed Ethereum sharding architecture). This
  allows to build L1 chains designed explicitly around their application and value proposition.
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+- **Shared security and financial scalability of L1 chains**. Any L1 chain attached to a Polkadot
+  core (and any other L2 solution to scale that L1) will benefit from Polkadot shared security
+  model. This means the Polkadot Nominated-Proof-of-Stake (NPoS) mechanism will secure L1 and L2
+  chains without them worrying about having enough proof-of-stake.
```
L2 security assumptions differ from parachain security assumptions. By design. all the parachains enjoy the same levels of security guarantees provided by Polkadot. The L2 solutions developed on top of L1s do not, as they do not adhere to any particular standard.

```suggestion
- **Shared security and financial scalability of L1 chains**. Any L1 chain attached to a Polkadot
  core can benefit from Polkadot shared security
  model. This means the Polkadot Nominated-Proof-of-Stake (NPoS) mechanism along with its consensus mechanism, will secure L1
  chains out-of-the-box without having to bootstrap security on their own.
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+  core (and any other L2 solution to scale that L1) will benefit from Polkadot shared security
+  model. This means the Polkadot Nominated-Proof-of-Stake (NPoS) mechanism will secure L1 and L2
+  chains without them worrying about having enough proof-of-stake.
+- **Secure interoperability.** Any L1 chain attached to Polkadot will benefit from Polkadot's native
```
```suggestion
- **Secure interoperability.** Any L1 chain attached to Polkadot (as well as L2 chains built on top of them) can benefit from Polkadot's native
```

Commit:
minor edits

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+  compromising scalability and throughput, and through on-chain treasury funds that can be accessed
+  through governance referendum. Those funds guarantee constant sponsorship for events, initiatives,
+  educational material, software development, etc., without any central entity being able to
+  interfere.
```
```suggestion
  education, software development, etc.
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+  through governance referendum. Those funds guarantee constant sponsorship for events, initiatives,
+  educational material, software development, etc., without any central entity being able to
+  interfere.
+- **Fast deployment of L1 chains.** This is achieved through the modular and flexible
```
```suggestion
- **Fast development and deployment of L1 chains.** This is achieved through the modular and flexible
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+
+### Polkadot's Representation
+
+The Polkadot realy-chain has been represented as a central ring surrounded by multiple parachains
```
```suggestion
The Polkadot realy-chain has been represented as a relay chain surrounded by multiple parachains
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+[cross-chain messaging format (XCM)](./learn-xcm-index) allowing messages to be delivered across
+chains having different logic.
+
+### Polkadot's Additional Tasks
```
```suggestion
### Polkadot's Additional Functionalities
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+### Decentralization as a Tool
+
+Decentralization is a crucial aspect when building a blockchain, but there must be a trade-off
+between:
```
```suggestion
Decentralization is a crucial aspect of blockchain networks, but there is a trade-off
between:
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+between:
+
+- having an over-decentralized network that struggles to reach consensus and consumes a lot of
+  energy to stay online, and
```
```suggestion
  energy to operate, and
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+- having an over-decentralized network that struggles to reach consensus and consumes a lot of
+  energy to stay online, and
+- having an under-decentralized network that reaches consensus fast at the expense of being
+  centralized and trivial to attack.
```
```suggestion
- having a network that reaches consensus fast at the expense of being
  centralized, making it trivial to manipulate or attack.
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+- having an under-decentralized network that reaches consensus fast at the expense of being
+  centralized and trivial to attack.
+
+Ideally, a network should be decentralized enough to make it practically impossible for someone to
```
```suggestion
Ideally, a network should be decentralized "enough" to make it practically impossible for someone to
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+  centralized and trivial to attack.
+
+Ideally, a network should be decentralized enough to make it practically impossible for someone to
+stop all nodes and take the network down. So, decentralization is a tool while the goal is
```
```suggestion
exert manipulative or malicious influence on the network. So, decentralization is a tool while the goal is
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+Ideally, a network should be decentralized enough to make it practically impossible for someone to
+stop all nodes and take the network down. So, decentralization is a tool while the goal is
+resilience, which is achieved by additionally providing on-chain treasury and governance mechanism
+allowing continuous financial incentives for the network's participants without intermediaries or
```
```suggestion
allowing continuous incentives for the network's participants without relying on intermediaries or
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+stop all nodes and take the network down. So, decentralization is a tool while the goal is
+resilience, which is achieved by additionally providing on-chain treasury and governance mechanism
+allowing continuous financial incentives for the network's participants without intermediaries or
+centralized entities being able to stop the network's progress.
```
```suggestion
centralized entities.
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+allowing continuous financial incentives for the network's participants without intermediaries or
+centralized entities being able to stop the network's progress.
+
+Currently, Polkadot 1.0 achieve resilience through the followings:
```
```suggestion
Currently, Polkadot 1.0 achieve resilience through the following strategies:
```

profradha commented on docs/general/polkadot-vision.md:
```diff
@@ -0,0 +1,194 @@
+
+Currently, Polkadot 1.0 achieve resilience through the followings:
+
+- The stake per validator is maximized and evenly distributed across validators.
```
```suggestion
- Nominated Proof of Staking (NPoS) where the stake per validator is maximized and evenly distributed across validators.
```

DrW3RK reviewed (commented):
Rephrased a bit. Have not gotten to the Blockspace section yet. Will review it later today or next week.

Commit:
Radha's feedback Polkadot 1.0

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+slug: ../polkadot-v1
+---
+
+## A Multi-chain Future
```
Is it possible we could combine the two sections, as Polkadot 1.0 is the Multi-chain future?  i.e.,

```
## Polkadot 1.0: A Multi-chain Future
```

filippoweb3 replied:
I do not fully agree with this as a multi-chain future can exist without polkadot. I mean, multi-chain is bigger than polkadot, polkadot just provides a way towards it. Does this makes sense?

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+initiatives:
+
+- **Application-specific Layer-1 (L1) blockchains (or parachains).** Polkadot is a sharded network
+  where transactions are processed parallelly within each shard. Polkadot shards can be heterogenous
```
```suggestion
  where transactions are processed in parallel with each shard. Polkadot shards can be heterogenous
```

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+
+### Polkadot's Resilience
+
+Decentralization is a crucial aspect of blockchain networks, but there is a trade-off between:
```
I'm not sure if it would be good to mention the trilemma here? (decentralization vs. scalability vs. security)

filippoweb3 replied:
Yep, I can briefly mention it (although I think the trilemma is over-cited by so many projects that it is not so interesting anymore IMO)

CrackTheCode016 replied:
Yeah true, that's why I wasn't sure if it was even interesting to even address 😅

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+  participants and further increase physical (how many validator nodes per service provider) and
+  social decentralization (how many validator nodes per operator). Those can be explored with the
+  [Polkawatch App](https://polkadot.polkawatch.app/).
+- An on-chain treasury and governance where every decision goes though public referenda and any
```
```suggestion
- An on-chain treasury and governance (see: [OpenGov](../learn/learn-polkadot-opengov.md) where every decision goes though public referenda and any
```

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+Habermeier as well as [this article](https://www.rob.tech/polkadot-blockspace-over-blockchains/) by
+him.
+
+### A Perspective Shift
```
```suggestion
### A Perspective Shift: Polkadot 2.0
```

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+being tamper-proof, public permissionless blockchains like Polkadot store data that are visible to
+everybody (i.e. public), and anybody can become a network participant (i.e. permissionless).
+
+So, the raw material produced by blockchains is blockspace; and how blockspace is organized,
```
```suggestion
The raw material produced by blockchains is **blockspace**; and how blockspace is organized,
```

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+accessed, and used will eventually shape the degree of scalability of the network. There are three
+distinctive blockspace features:
+
+- **security**, intended as how secure the blockspace is. In Proof-of-Stake (PoS) networks, this is
```
```suggestion
- **Security**, intended as how secure the blockspace is. In Proof-of-Stake (PoS) networks, this is
```

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+  operators (this will determine the degree of social centralization of the network), and how many
+  validators run on the same service provider (this will determine the degree of physical
+  centralization of the network).
+- **flexibility**, intended as how flexible the blockspace is, what can be done with it, and what
```
```suggestion
- **Flexibility**, intended as how flexible the blockspace is, what can be done with it, and what
```

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+  type of data can be stored. Data quality plays an important role depending on the type of network.
+  One might avoid having situations in which poor quality data flood blockspace hindering the prompt
+  execution of vital processes.
+- **availability**, intended as how available blockspace is and how difficult it is to access it. It
```
```suggestion
- **Availability**, intended as how available blockspace is and how difficult it is to access it. It
```

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+The design and realization of Polkadot 1.0 allowed its creators to truly realize the importance of
+blockspace.
+
+A blockchain is a way to store data. The storage unit is the block, and once a block is finalized
```
```suggestion
Blockspace represents the raw material that is produced as a result of the secure, trustless compute that blockchains provide.  A blockchain defines a way to propagate a set of state changes, where the storage unit for these state changes is the block, and once a block is finalized
```

filippoweb3 replied:
Blockspace can be non-secure and non-trustless. So I do not fully agree that blockspace "is produced as a result of secure and trustless computation". Do you also agree? I can produce a block with some data myself without a blockchain.

CrackTheCode016 replied:
Of course, but if we're talking about Polkadot's specific goal, then Polkadot provides that.  I can edit it to reflect that particular point

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+  secondary market options to ensure the usage of "second-hand" blockspace.
+
+Polkadot's has been designed around those core blockspace principles. However, its design can be
+further improved. Tasks such as balances transfers, staking, governance, etc. can be delegated to
```
```suggestion
further improved. Tasks which are currently managed on the relay chain, such as balances transfers, staking, and governance, can be delegated to
```

CrackTheCode016 commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,154 @@
+could become.
+
+Enter Polkadot 2.0, a long-term vision looking at Polkadot as a decentralized, secure, ubiquitous
+computing engine to power the next-generation of Web3 application.
```
A small excerpt here could be warranted, let me know what you think. Feel free to modify it, it is very rough.  I want to capture the concepts of "spaces", cores, etc: 

> In the context of Polkadot 2.0, the perspective shift will be crucial to realize the broader ideas it will bring due to its implementation.  From blockchains, a relatively two-dimensional structure, to defining governable, three-dimensional spaces of secure, decentralized computing.  This space would comprise multiple cores, similar to how a modern CPU is structured.  These cores can then communicate using a common language, XCM, which empowers applications built across these cores.   The goal is not a chain-centric future but an application-centric one that provides users true value and utility.

CrackTheCode016 reviewed (commented):
My initial comments, let me know if you would like me to expound on any points in particular.

filippoweb3 committed:
Update docs/general/polkadot-v1.md

Co-authored-by: bader y <ibnbassem@gmail.com>

filippoweb3 committed:
Update docs/general/polkadot-v1.md

Co-authored-by: bader y <ibnbassem@gmail.com>

filippoweb3 committed:
Update docs/general/polkadot-v1.md

Co-authored-by: bader y <ibnbassem@gmail.com>

filippoweb3 committed:
Update docs/general/polkadot-v1.md

Co-authored-by: bader y <ibnbassem@gmail.com>

filippoweb3 committed:
Update docs/general/polkadot-v1.md

Co-authored-by: bader y <ibnbassem@gmail.com>

filippoweb3 committed:
Update docs/general/polkadot-v1.md

Co-authored-by: bader y <ibnbassem@gmail.com>

filippoweb3 committed:
Update docs/general/polkadot-v1.md

Co-authored-by: bader y <ibnbassem@gmail.com>

filippoweb3 committed:
Update docs/general/polkadot-v1.md

Co-authored-by: bader y <ibnbassem@gmail.com>

DrW3RK committed:
remove title

The content in the section can serve as a preface to Polkadot 1.0 and need not have a title of its own

DrW3RK committed:
Remove Polkadot 2.0 references

profradha commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,153 @@
+## Polkadot 1.0
+
+Polkadot 1.0 reflects the status of Polkadot in 2023 at time of the release of the Polkadot runtime
+v1.0. This sections focuses on Polkadot 1.0 and some philosophical digressions about network
```
```suggestion
Polkadot 1.0 reflects the status of Polkadot in 2023 at time of the release of the [Polkadot runtime
v1.0.0](https://github.com/paritytech/polkadot/releases/tag/v1.0.0). This sections focuses on Polkadot 1.0 and some philosophical digressions about network
```

DrW3RK committed:
Update docs/general/polkadot-v1.md

DrW3RK committed:
Update polkadot-v1.md

profradha commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,152 @@
+  execution of vital processes.
+- **Availability**, intended as how available blockspace is and how difficult it is to access it. It
+  should not be too difficult to get your hands on it so that any business model (not only DeFi) can
+  thrive using it. Ideally, a marketplace must drive the blockspace price based on demand, with
```
```suggestion
  should not be too difficult to get your hands on it so that any business model  can
  thrive using it. Ideally, a marketplace must drive the blockspace price based on demand, with
```

profradha commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,152 @@
+  thrive using it. Ideally, a marketplace must drive the blockspace price based on demand, with
+  secondary market options to ensure the usage of "second-hand" blockspace.
+
+Polkadot's has been designed around those core blockspace principles. However, its design can be
```
```suggestion
Polkadot has been designed around those core blockspace principles. However, its design can be
```

profradha commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,152 @@
+  secondary market options to ensure the usage of "second-hand" blockspace.
+
+Polkadot's has been designed around those core blockspace principles. However, its design can be
+further improved. Tasks which are currently managed on the relay chain, such as balances transfers, staking, and governance, can be delegated to
```
```suggestion
further improved such that the tasks which are currently managed on the relay chain, such as balances transfers, staking, and governance, can be delegated to
```

profradha commented on docs/general/polkadot-v1.md:
```diff
@@ -0,0 +1,152 @@
+Habermeier as well as [this article](https://www.rob.tech/polkadot-blockspace-over-blockchains/) by
+him.
+
+### A Perspective Shift: Upcoming Polkadot Feature Developments
```
```suggestion
### A Perspective Shift: Upcoming Polkadot Features Development
```

DrW3RK committed:
Update docs/general/polkadot-v1.md

DrW3RK committed:
Update docs/general/polkadot-v1.md

DrW3RK committed:
Update docs/general/polkadot-v1.md

DrW3RK committed:
Update docs/general/polkadot-v1.md
